src/test/scala/akka/persistence/cassandra/query/EventsByTagSpec.scala

```diff
@@ -817,3 +820,34 @@ class EventsByTagStrictBySeqNoSpec extends AbstractEventsByTagSpec("EventsByTagS
 
 }
 
+class EventsByTagStrictBySeqNoEarlyFirstOffsetSpec
+  extends AbstractEventsByTagSpec("EventsByTagStrictBySeqNoEarlyFirstOffsetSpec", EventsByTagSpec.strictConfigFirstOffset1001DaysAgo) {
+  import EventsByTagSpec._
+
+  "Cassandra live eventsByTag with delayed-event-timeout > 0s and firstOffset = 1000 days ago" must {
+    "find all events when starting the query 1000 days ago" in {
+      val t1 = LocalDateTime.now(ZoneOffset.UTC).minusMinutes(5).minusDays(1001)
+      val t2 = t1.minusMinutes(1)
+      val w1 = UUID.randomUUID().toString
+      val w2 = UUID.randomUUID().toString
+
+      // create two events per day over the last 1000 days to be sure that delayed event backtracking is
+      // triggered before reaching the current day timebucket
+      (1L to 1000L).foreach { n =>
+        val eventA = PersistentRepr(s"A$n", n, "a", "", writerUuid = w1)
+        val eventB = PersistentRepr(s"B$n", n, "b", "", writerUuid = w2)
+        writeTestEvent(t1.plus(n, ChronoUnit.DAYS), eventA, Set("T11"))
+        writeTestEvent(t2.plus(n, ChronoUnit.DAYS), eventB, Set("T11"))
+      }
+
+      // the search for delayed events should start before we get to the current timebucket
+      // until 0.26/0.51 backtracking was broken and events would be skipped
+      val src = queries.eventsByTag(tag = "T11", offset = NoOffset)
+      val probe = src.runWith(TestSink.probe[Any])
+      probe.request(2000)
+      probe.expectNextN(2000)
+      probe.cancel()
+    }
```

Comment:
Good test, it didn't fail for me (reverted the change in EventsByTagPublisher), but it's a timing issue as you said. If the `LookingForDelayed` kicks in before the initial looping of time buckets have finished it will be wrong. Reproduced with a strategic sleep.

```
[DEBUG] [04/24/2017 15:02:29.755] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-5] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query for tag [T11] from [20170415] [e92e8470-2111-11e7-8406-d74cacf80b1d (2017-04-14 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:29.781] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-6] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query chunk done for tag [T11], timBucket [20170415], count [2] of [2]
[DEBUG] [04/24/2017 15:02:29.781] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-6] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query for tag [T11] from [20170416] [13984470-21db-11e7-8406-d74cacf80b1d (2017-04-15 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:29.809] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query chunk done for tag [T11], timBucket [20170416], count [2] of [2]
[DEBUG] [04/24/2017 15:02:29.809] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query for tag [T11] from [20170417] [3e020470-22a4-11e7-8406-d74cacf80b1d (2017-04-16 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:29.835] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query chunk done for tag [T11], timBucket [20170417], count [2] of [2]
[DEBUG] [04/24/2017 15:02:29.835] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query for tag [T11] from [20170418] [686bc470-236d-11e7-8406-d74cacf80b1d (2017-04-17 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:29.860] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query chunk done for tag [T11], timBucket [20170418], count [2] of [2]
[DEBUG] [04/24/2017 15:02:29.862] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query for tag [T11] from [20170418] [8fda93f0-2436-11e7-8080-808080808080 (2017-04-18 12:57:21)] limit [50]
[DEBUG] [04/24/2017 15:02:29.889] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-6] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query chunk done for tag [T11], timBucket [20170418], count [0] of [1]
[DEBUG] [04/24/2017 15:02:29.889] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-6] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query for tag [T11] from [20170419] [92d58470-2436-11e7-8406-d74cacf80b1d (2017-04-18 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:29.912] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query chunk done for tag [T11], timBucket [20170419], count [0] of [0]
[DEBUG] [04/24/2017 15:02:29.912] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query for tag [T11] from [20170420] [92d58470-2436-11e7-8406-d74cacf80b1d (2017-04-18 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:29.937] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query chunk done for tag [T11], timBucket [20170420], count [0] of [0]
[DEBUG] [04/24/2017 15:02:29.937] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query for tag [T11] from [20170421] [92d58470-2436-11e7-8406-d74cacf80b1d (2017-04-18 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:29.964] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query chunk done for tag [T11], timBucket [20170421], count [0] of [0]
[DEBUG] [04/24/2017 15:02:29.964] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query for tag [T11] from [20170422] [92d58470-2436-11e7-8406-d74cacf80b1d (2017-04-18 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:29.987] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query chunk done for tag [T11], timBucket [20170422], count [0] of [0]
[DEBUG] [04/24/2017 15:02:29.987] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query for tag [T11] from [20170423] [92d58470-2436-11e7-8406-d74cacf80b1d (2017-04-18 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:30.011] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query chunk done for tag [T11], timBucket [20170423], count [0] of [0]
[DEBUG] [04/24/2017 15:02:30.011] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query for tag [T11] from [20170424] [92d58470-2436-11e7-8406-d74cacf80b1d (2017-04-18 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:30.036] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-7] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] backtracking delayed query chunk done for tag [T11], timBucket [20170424], count [0] of [0]
[DEBUG] [04/24/2017 15:02:30.376] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-6] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query for tag [T11] from [20170424] [92d58470-2436-11e7-8406-d74cacf80b1d (2017-04-18 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:30.403] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-8] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query chunk done for tag [T11], timBucket [20170424], count [0] of [0]
[DEBUG] [04/24/2017 15:02:30.866] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-6] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query for tag [T11] from [20170424] [92d58470-2436-11e7-8406-d74cacf80b1d (2017-04-18 12:57:26)] limit [50]
[DEBUG] [04/24/2017 15:02:30.891] [EventsByTagStrictBySeqNoEarlyFirstOffsetSpec-cassandra-plugin-default-dispatcher-8] [akka://EventsByTagStrictBySeqNoEarlyFirstOffsetSpec/user/StreamSupervisor-0/flow-2-2-actorPublisherSource] query chunk done for tag [T11], timBucket [20170424], count [0] of [0]
```